### PR TITLE
[HIPIFY][Tensor][feature] hipTensor support - Functions - Part 9

### PR DESCRIPTION
**Pull request by aakanksha555**

### CODE REVIEW

File: src/CUDA2HIP_TENSOR_API_types.cpp [hunk 1]
```diff
@@ -175,7 +175,8 @@ const std::map<llvm::StringRef, hipCounter> CUDA_TENSOR_TYPE_NAME_MAP {
   {"CUTENSOR_PLAN_REQUIRED_WORKSPACE",                 {"",                                                         "",           CONV_NUMERIC_LITERAL, API_TENSOR, 1, UNSUPPORTED}},
 
   {"cutensorHandle_t",                                 {"hiptensorHandle_t",                                        "",           CONV_TYPE, API_TENSOR, 1}},
-  {"cutensorTensorDescriptor",                         {"hiptensorTensorDescriptor_t",                              "",           CONV_TYPE, API_TENSOR, 1}},
+  {"cutensorTensorDescriptor_t",                       {"hiptensorTensorDescriptor_t",                              "",           CONV_TYPE, API_TENSOR, 1}},
```
Comment (emankov) [inline src/CUDA2HIP_TENSOR_API_types.cpp:179]: `cutensorTensorDescriptor` and `cutensorTensorDescriptor_t` both declared in cuTensor:
```cpp
typedef struct cutensorTensorDescriptor *cutensorTensorDescriptor_t;
```
Mark `cutensorTensorDescriptor` as `UNSUPPORTED` for now.

Additionally, please revise the rest struct/enum declarations with typedefs.

File: src/CUDA2HIP_TENSOR_API_functions.cpp [hunk 2]
```diff
@@ -30,9 +30,18 @@ const std::map<llvm::StringRef, hipCounter> CUDA_TENSOR_FUNCTION_MAP {
   {"cutensorHandleReadPlanCacheFromFile",                         {"",                                                    "",                                     CONV_LIB_FUNC, API_TENSOR, 2, UNSUPPORTED}},
   {"cutensorWriteKernelCacheToFile",                              {"",                                                    "",                                     CONV_LIB_FUNC, API_TENSOR, 2, UNSUPPORTED}},
   {"cutensorReadKernelCacheFromFile",                             {"",                                                    "",                                     CONV_LIB_FUNC, API_TENSOR, 2, UNSUPPORTED}},
-  {"cutensorCreateTensorDescriptor",                              {"",                                                    "",                                     CONV_LIB_FUNC, API_TENSOR, 2, UNSUPPORTED}},
+  {"cutensorCreateTensorDescriptor",                              {"hiptensorInitTensorDescriptor",                       "",                                     CONV_LIB_FUNC, API_TENSOR, 2, HIP_UNSUPPORTED}},
```
Comment (emankov) [inline src/CUDA2HIP_TENSOR_API_functions.cpp]: Let's use `UNSUPPORTED` for Tensor, as far as we don't have the corresponding `ROC` component for `cuTensor`.

File: tests/unit_tests/synthetic/libraries/cutensor2hiptensor.cu [hunk 3]
```diff
@@ -16,16 +16,31 @@ int main() {
   cutensorStatus_t status;
 
   //CHECK: hiptensorTensorDescriptor_t *tensorDescriptor;
-  cutensorTensorDescriptor *tensorDescriptor;
+  //CHECK: hiptensorTensorDescriptor_t *descA;
+  //CHECK: hiptensorTensorDescriptor_t *descB;
+  cutensorTensorDescriptor_t *tensorDescriptor;
+  cutensorTensorDescriptor_t *descA;
+  cutensorTensorDescriptor_t *descB;
```
Comment (emankov) [inline tests/unit_tests/synthetic/libraries/cutensor2hiptensor.cu]: Initialization is desirable.

File: tests/unit_tests/synthetic/libraries/cutensor2hiptensor.cu [hunk 4]
```diff
@@ -16,16 +16,31 @@ int main() {
   cutensorStatus_t status;
 
   //CHECK: hiptensorTensorDescriptor_t *tensorDescriptor;
-  cutensorTensorDescriptor *tensorDescriptor;
+  //CHECK: hiptensorTensorDescriptor_t *descA;
+  //CHECK: hiptensorTensorDescriptor_t *descB;
```
Comment (emankov) [inline tests/unit_tests/synthetic/libraries/cutensor2hiptensor.cu]: CHECK-NEXT

**Review by emankov — CHANGES_REQUESTED** [review]

File: src/CUDA2HIP_TENSOR_API_functions.cpp (hunk 2 above)
Comment (aakanksha555) [reply src/CUDA2HIP_TENSOR_API_functions.cpp]: Done.

**Review by emankov — APPROVED** [review]
LGTM, thanks!

**Pull request merged**